### PR TITLE
feat(runtimes): Deno 1.44.2

### PR DESCRIPTION
**Pull request by janpio**
Deno fixed things, and Prisma now works!

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -1696,10 +1740,7 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-
-      - name: Define Client Engine Type to test
-        run: echo "PRISMA_CLIENT_ENGINE_TYPE=${{ matrix.clientEngine }}" >> $GITHUB_ENV
-
+
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion
      
      - name: Define Client Engine Type to test
        run: echo "PRISMA_CLIENT_ENGINE_TYPE=${{ matrix.clientEngine }}" >> $GITHUB_ENV
        
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -1708,9 +1749,8 @@ jobs:
         with:
           node-version: 16
           cache: 'pnpm'
-          cache-dependency-path: ${{ github.job }}/${{ matrix.feature }}/pnpm-lock.yaml
-
-      - name: Install Dependencies
+          
+     - name: Install Dependencies
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion
          cache-dependency-path: ${{ github.job }}/${{ matrix.feature }}/pnpm-lock.yaml
     
     - name: Install Dependencies
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -1732,6 +1772,25 @@ jobs:
...
+        uses: nick-invision/retry@v3
+        with:
+          timeout_minutes: 10
+          max_attempts: 3
+          command: bash .github/scripts/test-project.sh ${{ github.job }} ${{ matrix.feature }}
+
+      - name: notify-slack
+        if: failure()
+        run: bash .github/slack/notify-failure.sh ${{ github.job }} ${{ matrix.feature }}
+
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion
```

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -1696,10 +1740,10 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-
+
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion

```

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -1696,10 +1740,10 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-
+      
       - name: Define Client Engine Type to test
         run: echo "PRISMA_CLIENT_ENGINE_TYPE=${{ matrix.clientEngine }}" >> $GITHUB_ENV
-
+
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion

```

File: .github/workflows/test.yaml [hunk 6]
```diff
@@ -1708,9 +1752,10 @@ jobs:
         with:
           node-version: 16
           cache: 'pnpm'
+
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion
```

File: .github/workflows/test.yaml [hunk 7]
```diff
@@ -1708,9 +1752,10 @@ jobs:
         with:
           node-version: 16
           cache: 'pnpm'
+          
           cache-dependency-path: ${{ github.job }}/${{ matrix.feature }}/pnpm-lock.yaml
-
-      - name: Install Dependencies
+     
+     - name: Install Dependencies
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion

      - name: Install Dependencies
```

File: .github/workflows/test.yaml [hunk 8]
```diff
@@ -1696,10 +1740,9 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-
+      
       - name: Define Client Engine Type to test
         run: echo "PRISMA_CLIENT_ENGINE_TYPE=${{ matrix.clientEngine }}" >> $GITHUB_ENV
```
Comment (janpio) [inline .github/workflows/test.yaml:1754]:
```suggestion
        run: echo "PRISMA_CLIENT_ENGINE_TYPE=${{ matrix.clientEngine }}" >> $GITHUB_ENV

```

File: .github/workflows/test.yaml [hunk 9]
```diff
@@ -1682,6 +1685,47 @@ jobs:
...
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 16
+          cache: 'pnpm'
+          cache-dependency-path: ${{ github.job }}/${{ matrix.runtime }}/pnpm-lock.yaml
+
+      - name: use deno v1.x
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: v1.44.2
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion
  runtimes:
    needs: [detect_jobs_to_run]
    if: contains(fromJSON(needs.detect_jobs_to_run.outputs.jobs), 'runtimes')

    timeout-minutes: 10
    strategy:
      fail-fast: false
      matrix:
        runtime:
          - deno
          #- bun
        clientEngine: ['library', 'binary']
        os: [ubuntu-20.04]
    runs-on: ${{ matrix.os }}
    concurrency: ${{ github.job }}-${{ matrix.runtime }}-${{ matrix.clientEngine }}
    env:
      RUNTIMES_DENO_DATABASE_URL: ${{ secrets.DATAPROXY_COMMON_URL_SOURCE }}

    steps:
      - uses: actions/checkout@v4

      - uses: pnpm/action-setup@v3.0.0
        with:
          version: 8

      - uses: actions/setup-node@v4
        with:
          node-version: 16
          cache: 'pnpm'
          cache-dependency-path: ${{ github.job }}/${{ matrix.runtime }}/pnpm-lock.yaml

      - name: use deno v1.x
        uses: denoland/setup-deno@v1
        with:
          deno-version: v1.44.2

      - name: Define Client Engine Type to test
        run: echo "PRISMA_CLIENT_ENGINE_TYPE=${{ matrix.clientEngine }}" >> $GITHUB_ENV

      - name: Install Dependencies
        run: pnpm install

      - name: test ${{ matrix.runtime }}
        run: bash .github/scripts/test-project.sh ${{ github.job }} ${{ matrix.runtime }}

      - name: notify-slack
        if: failure()
        run: bash .github/slack/notify-failure.sh ${{ github.job }} ${{ matrix.runtime }}
```

File: runtimes/deno/prisma/schema.prisma [hunk 10]
```diff
@@ -0,0 +1,34 @@
+generator client {
+  provider   = "prisma-client-js"
+  output     = "../generated/client"
+  engineType = "binary" // overwritten when run in CI via env var
+  // previewFeatures = ["deno"]
```
Comment (Jolg42) [inline runtimes/deno/prisma/schema.prisma]:
```suggestion
```

Why is the preview feature commented here?

Comment (Jolg42) [reply runtimes/deno/prisma/schema.prisma]: add a comment if necessary maybe

File: runtimes/deno/src/run.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { type PrismaClient } from '../generated/client/index.d.ts'
+import { createRequire } from 'node:module'
+const require = createRequire(import.meta.url)
+const Prisma = require('../generated/client/index.js')
+
+// const prisma = new PrismaClient();
+const prisma: PrismaClient = new Prisma.PrismaClient({ log: ['query', 'info', 'warn', 'error'] })
+
+console.log("foo")
```
Comment (Jolg42) [inline runtimes/deno/src/run.ts]:
```suggestion
```

**Review by Jolg42 — APPROVED** [review]

**Pull request merged**